probe-rs/src/core.rs

```diff
@@ -958,7 +911,32 @@ pub enum ResolvedCoreOptions {
         options: XtensaCoreAccessOptions,
     },
 }
+
 impl ResolvedCoreOptions {
+    fn new(target: &Target, options: CoreAccessOptions) -> Self {
+        match options {
+            CoreAccessOptions::Arm(options) => {
+                if let DebugSequence::Arm(sequence) = target.debug_sequence.clone() {
+                    return ResolvedCoreOptions::Arm { sequence, options };
+                }
+            }
+            CoreAccessOptions::Riscv(options) => {
+                if let DebugSequence::Riscv(sequence) = target.debug_sequence.clone() {
+                    return ResolvedCoreOptions::Riscv { sequence, options };
+                }
+            }
+            CoreAccessOptions::Xtensa(options) => {
+                if let DebugSequence::Xtensa(sequence) = target.debug_sequence.clone() {
```

Comment:
We could match on a tuple of (options, target.debug_sequence) to make this a bit nicer I think :)

Reply:
Looks a bit wide with inlay hints but yeah, good spot